### PR TITLE
Feature/add libraries table

### LINKED ISSUE #8: Table to list all libraries
## Description

blocked by #7 

Create a component `LibrariesTable` with only one column:
- Name
  
In LibrariesPage, load ten first libraries from backend and display it in table.

Here is the call we have to do: `api/classes/Library?limit=10`
See leto-modelizer for call headers.

image

### PR DESCRIPTION
**Pull request by k-y0u**
Resolves #8

### CODE REVIEW

File: src/composables/LetoModelizerApi.js [hunk 1]
```diff
@@ -38,3 +38,21 @@ export async function getUserRoles(userId, sessionToken) {
 
   return api.get(`/api/roles?${queryParameters}`, headers);
 }
+
+/**
+ * Get all libraries.
+ * @param {string} sessionToken - The current user's session token.
+ * @returns {Promise<object[]>} Return an array of libraries. Role is an object from ParseServer.
```
Comment (justinehell) [inline src/composables/LetoModelizerApi.js:45]: Role is... to remove

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/composables/LetoModelizerApi.js (hunk 1 above)
Comment (k-y0u) [reply src/composables/LetoModelizerApi.js]: done

File: tests/unit/composables/LetoModelizerApi.test.js [hunk 2]
```diff
@@ -55,4 +56,29 @@ describe('User Authentication', () => {
       expect(res.results).toEqual([{ name: 'admin' }]);
     });
   });
+
+  describe('Test function: getLibraries', () => {
+    it('should return the current user roles', async () => {
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply tests/unit/composables/LetoModelizerApi.test.js]: done

**Review by justinehell — APPROVED** [review]

**Review by aboubakar14 — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

File: src/composables/LetoModelizerApi.js [hunk 3]
```diff
@@ -38,3 +38,21 @@ export async function getUserRoles(userId, sessionToken) {
 
   return api.get(`/api/roles?${queryParameters}`, headers);
 }
+
+/**
+ * Get all libraries.
+ * @param {string} sessionToken - The current user's session token.
+ * @returns {Promise<object[]>} Return an array of libraries.
+ */
+export async function getLibraries(sessionToken) {
```
Comment (Zorin95670) [inline src/composables/LetoModelizerApi.js:47]: you can remove `sessionToken` from parameters and put use directly method to get it in the function.

File: src/pages/LibrariesPage.vue [hunk 4]
```diff
@@ -1,10 +1,44 @@
...
-  <q-page class="flex q-pl-md bg-grey-1">
+  <q-page class="column q-pa-md bg-grey-1">
     <h4
-      class="q-ma-none q-mt-md"
+      class="q-ma-none q-mt-md q-mb-lg"
       data-cy="page_libraries_title"
     >
       {{ $t('LibrariesPage.text.title') }}
     </h4>
+    <q-table
```
Comment (Zorin95670) [inline src/pages/LibrariesPage.vue:9]: Put table in a LibrariesTable like the issues says with column setup and other

File: changelog.md [hunk 5]
```diff
@@ -16,3 +16,4 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Setup Libraries page
 * Setup Roles page
 * Setup User Groups page
+* Add Libraries table
```
Comment (Zorin95670) [inline changelog.md:19]: modidfy like this:
```
* Setup libraries page with:
  * Table to display all libraries
```

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: changelog.md (hunk 5 above)
Comment (k-y0u) [reply changelog.md]: done

File: src/pages/LibrariesPage.vue (hunk 4 above)
Comment (k-y0u) [reply src/pages/LibrariesPage.vue]: done

File: src/composables/LetoModelizerApi.js (hunk 3 above)
Comment (k-y0u) [reply src/composables/LetoModelizerApi.js]: done

File: src/components/tables/LibrariesTable.vue [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    data-cy="page_libraries_table"
+  />
+</template>
+
+<script setup>
+import { onMounted, ref } from 'vue';
+import { getLibraries } from 'src/composables/LetoModelizerApi';
+
+const pagination = ref({
+  rowsPerPage: 0, // infinite
```
Comment (Zorin95670) [inline src/components/tables/LibrariesTable.vue:19]: 10

File: src/components/tables/LibrariesTable.vue [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+});
+const columns = ref([{
+  name: 'name',
+  required: true,
+  label: 'Name',
+  align: 'left',
+  field: 'name',
+}]);
+const libraries = ref([]);
+const rowKey = ref('name');
```
Comment (Zorin95670) [inline src/components/tables/LibrariesTable.vue:29]: objectId

Comment (k-y0u) [reply src/components/tables/LibrariesTable.vue]: done

**Review by Zorin95670 — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Review by Zorin95670 — CHANGES_REQUESTED** [review]
changelog missing

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**